src/mongocxx/exception/base.hpp

```diff
@@ -30,33 +27,50 @@ namespace mongocxx {
 MONGOCXX_INLINE_NAMESPACE_BEGIN
 namespace exception {
 
-using error_and_code_type = std::tuple<std::string, std::int32_t>;
-
+///
+/// A class to be used as the base class for all mongocxx exceptions.
+///
 class MONGOCXX_API base : virtual public std::exception {
```

Comment:
Well, the bsoncxx base exception definitely doesn't need the server error in its base class. I'm also not convinced that we won't later have exceptions that are self generated, and therefore don't have a server error at all.

My feeling right now on the error handling stuff is that we need to step back and do some design. I think it is the only open area of work that we don't have a solid answer on yet.

So, lets put this one on hold and try to move the other work forward with the exception system we have (which I admit is not ideal).